src/native/libs/System.Globalization.Native/pal_locale.m

```diff
@@ -786,13 +786,21 @@ int32_t GlobalizationNative_GetLocalesNative(UChar* value, int32_t length)
     }
 }
 
+static NSString* GetBaseName(NSString *localeIdentifier)
+{
+    NSLocale *locale = [[NSLocale alloc] initWithLocaleIdentifier:localeIdentifier];
+    NSString *languageCode = [locale objectForKey:NSLocaleLanguageCode];
+    return languageCode;
+}
+
 const char* GlobalizationNative_GetDefaultLocaleNameNative(void)
 {
     @autoreleasepool
     {
         if (NSLocale.preferredLanguages.count > 0)
         {
-            return strdup([NSLocale.preferredLanguages[0] UTF8String]);
+            NSString *preferredLanguage = [NSLocale.preferredLanguages objectAtIndex:0];
+            return strdup([GetBaseName(preferredLanguage) UTF8String]);
```

Comment:
Why do we need to return only the language code instead of the full locale code (including region)? I thought that locale is suppose to have both information.

Reply:
We need to return only the base name for the default locale, as ICU does. Reference: [link](https://github.com/dotnet/runtime/blob/117c4ab7723da207ad56014b625395c252af9b6a/src/native/libs/System.Globalization.Native/pal_locale.c#L232).
When both the language and region information are provided, it can cause exceptions. For example, **en-CZ** is not a valid culture code, so we should return just **en** in such cases.